"Sensitive data projects"-subchapter for Project Design (#2076)
alan-turing-institute/the-turing-way: book/website/project-design/sdpw/trew.md

The secure egress process and export policy for the TRE in question will necessarily be rigourous in proportion to the sensitivity of the data being analysed. 
It's therefore important to consider the time constraints associated with exporting any research outputs at the end of a project.
    
Developing research code within the TRE may be more appropriate where progress is overly slowed by developing externally, such as when data access is critical for any development. 
As such, it's important to consider the trade-off between developing research code within the TRE and outside of it.
    
**4) Do you aim to use notebooks (such as Jupyter) for analyses carried out in TREs?** 

Notebooks are a fantastic resource for data analysis, and this is no different in the context of working with sensitive data in a TRE. 
However, notebooks are a particularly risky for export from TRE. Jupyter notebook files in particular (.ipynb extension) are JSON documents that include metadata not viewable in the rendered form, which could increase the risk of sensitive data being included in any exported notebook by mistake.

> One recommendation when preparing the output of a data analysis carried out in a notebook within a TRE for publication is to convert it to a static format such as PDF.

Reviewer: Perhaps we can also add a link here to the "Sharing Your Jupyter Notebook" page

You may find TREs also described as "Data Safe Havens" or "Secure Research Environments" or other synonymous names, since the use of such platforms is relatively new and there is no strict consensus on the terminology. 
To learn more about what TREs are, check [this graphic](https://www.hdruk.ac.uk/access-to-health-data/trusted-research-environments/) produced by Health Data Research UK.

(pd-sdpw-trew-conducting)=
## Conducting research with TREs

In general, researchers working in TREs will find that much of the same guidance for working effectively and reproducibly as outlined in the [Guide for Reproducible Research](https://the-turing-way.netlify.app/reproducible-research/reproducible-research.html) section will be just as relevant as in any other computing environment.

There are however some important questions to consider when working in TREs that will affect the your development workflow for scientific code writing and data analysis:
    

> 1. Does the TRE have internet access?

Reviewer: ```suggestion
**1) Does the TRE have internet access?**
```

    
TREs set up for researchers to work on particularly sensitive data will likely lack internet access, in order to prevent a data breach or the accidental disclosure of sensitive data outside the TRE. 
Where this is the case, it is important to understand how the processes particular to your TRE work for things like software ingress (for important software libraries you may require) and code/data egress (to understand what, if anything, can be exported from the TRE). 
These policies will likely be set by those managing the TRE, for example the IT department at your University.
    

> 2. Does the TRE have a version control system, such as GitLab?

Reviewer: ```suggestion
**2) Does the TRE have a version control system, such as GitLab?**
```

    
Without internet access, websites like GitHub are unavailable for the development of research code within TREs. 
Version Control Systems (VCS) like Git can be useful even where access to GitHub is impossible. 
For example, [GitLab](https://about.gitlab.com/) can be used for code developed within the TRE, if it's installed in the environment. 

If the TRE does not have a collaboration/VCS tool such as GitLab installed, it may be preferable to develop the research code outside the TRE, bringing it into the TRE via secure code ingress at appropriate intervals. 
To learn more about version control, consult the [Guide for Reproducible Research](https://the-turing-way.netlify.app/reproducible-research/vcs.html).

> 3. Do you aim to publish research carried out on sensitive data in a TRE?

Reviewer: ```suggestion
**3) Do you aim to publish research carried out on sensitive data in a TRE?**
```

    
Developing research code outside the TRE, for example in a public GitHub repository, can also make publishing the research methods for TRE projects simpler. 
Developing code you wish to export from the TRE inside the environment carries risks. 
For example, sensitive data could be committed to the Git repository by mistake, which is impossible when the code has been developed in a separate location. 
The secure egress process and export policy for the TRE in question will necessarily be rigourous in proportion to the sensitivity of the data being analysed. 
It's therefore important to consider the time constraints associated with exporting any research outputs at the end of a project.
    
Developing research code within the TRE may be more appropriate where progress is overly slowed by developing externally, such as when data access is critical for any development. 
As such, it's important to consider the trade-off between developing research code within the TRE and outside of it.
    

> 4. Do you aim to use notebooks (such as Jupyter) for analyses carried out in TREs? 

Reviewer: ```suggestion
**4) Do you aim to use notebooks (such as Jupyter) for analyses carried out in TREs?** 
```

Notebooks are a fantastic resource for data analysis, and this is no different in the context of working with sensitive data in a TRE. 

> However, notebooks are a particularly risky for export from TRE, for the reasons outlined in the previous paragraph. 

Reviewer: ```suggestion
However, notebooks are a particularly risky for export from TRE. Jupyter notebook files in particular (.ipynb extension) are JSON documents that include metadata not viewable in the rendered form, which could increase the risk of sensitive data being included in any exported notebook by mistake.
```

One recommendation when preparing the output of a data analysis carried out in a notebook within a TRE for publication is to convert it to a static format such as PDF.
    

> 5. Do you want those without access to the TRE to be able to run the code used in your research?

Reviewer: ```suggestion
**5) Do you want those without access to the TRE to be able to run the code used in your research?**
```

    
Another suggestion could be to include a small synthetic dataset alongside any published code, which any publicly available code or notebooks containing the data analysis can be run with in lieu of access to the "real" (non-synthetic data) sensitive data from the TRE. 

> For more information on synthetic data, read the [synthetic data section]().

Reviewer: ```suggestion
```
